Attribute/WithLoggerChannel.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\MonologBundle\Attribute;
+
+/**
+ * Autoconfigures the logger channel used when inject a logger in the service.
+ *
+ * This adds the `monolog.logger` tag on the service definition.
+ */
+#[\Attribute(\Attribute::TARGET_CLASS)]
+final class WithLoggerChannel
+{
+    /**
+     * @readonly
+     */
+    public string $channel;
```

Comment:
```suggestion
    public $channel;
```

Reply:
IMO, it is fine to keep the typed property there. Attributes don't work on PHP 7.2 anyway

Reply:
ah indeed. For some reasons, I thought CPP was 8.1 like the readonly properties.